### PR TITLE
Update aws-sdk to v3

### PR DESCRIPTION
**Pull request by mwinckler**
- Use the compartmentalized version 3 of the AWS SDK instead of the monolithic version 2
- Remove dependency on sqs-consumer since it doesn't (yet?) support aws-sdk v3
- Bump package version to 0.1.0 to represent dependency version change

### CODE REVIEW

File: src/redriveDLQ.js [hunk 1]
```diff
@@ -1,20 +1,18 @@
...
 
 const sqs = require('./sqs')
 
-const sendMessage =
-  throat(16, promisify(sqs.sendMessage, sqs))
+const sendMessage = throat(16, sqs.sendMessage)
 
 const redriveDLQ = opts => dlq =>
-  new Promise((resolve, reject) => {
+  new Promise(async (resolve, reject) => {
```
Comment (mgreystone) [inline src/redriveDLQ.js:15]: The `async` inside a `Promise` seems like a code smell of mismatched patterns. Do we ned a promise constructor at all anymore & make a completely async function? Or alternatively, move the stuff that needs await into its own named function chained on the `.then` of this one?

**Review by mgreystone — COMMENTED** [review]
I gotta ask the needful. Sorry.

What is the problem we are trying to solve here with bumping the `aws-sdk`? Do we think it justifies the responsibility & risk of dropping & re-implementing `sqs-consumer`?

File: src/redriveDLQ.js [hunk 2]
```diff
@@ -64,22 +64,29 @@ const redriveDLQ = opts => dlq =>
...
-        .on('error', finish)
-        .on('processing_error', finish)
-        .on('timeout_error', finish)
-
-    consumer.start()
-
     const timer = setTimeout(tookTooLong, timeout)
+
+    // eslint-disable-next-line no-constant-condition
+    while (true) {
```
Comment (JWA111) [inline src/redriveDLQ.js:70]: I'm not a big fan of constant conditions. I think streaming would have been a better option here, which could be achieved using [highland](https://caolan.github.io/highland/) which is a relatively small package. I don't want to cause you a lot of rework, but if you feel inclined I'm happy to :pear: on it

**Comment by mwinckler** [comment]
@mgreystone No need to apologize! If this is crazy, I'm open to pushback and better alternatives.

My reasoning behind this change is that [team-media](https://github.com/articulate/team-media), a new service, is built using AWS SDK v3, which broke out the SDK packages into individual pieces so we only have to import what we need instead of the entire SDK. Because team-media queues are on EKS, existing botzo commands to redrive SQS queues don't work. Therefore I wanted to write yarn tasks to redrive queues and I was pointed to `overdrive` as an existing solution. But because `overdrive` still relies on AWS SDK v2, to reference it from team-media would mean importing the entire AWS SDK v2 in addition to the specific v3 packages the service code is using.

So I took it as an opportunity to move `overdrive` forward while keeping team-media packages slim. I considered the risk comparatively low given:

1) The package major version will be bumped, so no impact to existing consumers
2) The immediate use case is a repo still under development/not yet released to customers, so it's an ideal opportunity to test this/find any bugs before applying it to existing services.

If there's a better way to do this (or if, in your view, importing the v2 SDK isn't a big deal), I'm certainly open to direction.

**Comment by mgreystone** [comment]
Thanks for the explanation, @mwinckler .

First of, IMHO, importing the aws-sdk v2 alongside v3 isn't a big deal. I think my vote would be to wait until `sqs-consumer` supports v3.

> The package major version will be bumped, so no impact to existing consumers

This is true for this version. However, when `rise-api` needs 0.2.0 for a new feature someday, it will be forced to adopt this change as well & increase its scope of risk.

> The immediate use case is a repo still under development/not yet released to customers, so it's an ideal opportunity to test this/find any bugs before applying it to existing services.

This is a fair point. My only concern here would be if some service accidentally upgraded while `team-media` was still being developed (a minuscule risk, i admit).

What worries me the most is that _if_ this does introduce a problem, we probably won't know until during an active incident, which is when we use this lib the most.

---

All that said, v3 looks to be quite the improvement, and i don't want to stand in the way of progress. If we get some of these code style stuffs ironed out, i won't be the one to hold this up.

**Comment by mwinckler** [comment]
That's fair. We can wait on this until sqs-consumer supports v3, assuming it will eventually. I'll close this PR; thanks very much for taking a look at it!

**Comment by dpbackes** [comment]
rita

We need aws-sdk v3 for the purposes of getting this to just become an `art` command.

Will take a look at addressing the existing comments but lmk if there's anything else I should do

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "@articulate/overdrive",
-  "version": "0.0.4",
+  "name": "overdrive",
```
Comment (mgreystone) [inline package.json:2]: I don't think we can drop the prefix?

File: src/redriveDLQ.js (hunk 2 above)
Comment (mgreystone) [reply src/redriveDLQ.js:74]: I think do/while would read better, but i don't have a strong opinion here.

File: package.json (hunk 3 above)
Comment (dpbackes) [reply package.json:2]: ahh good catch, that's changed since Matt created this

**Pull request closed without merging**